BlueCapKit/Central/Peripheral.swift

```diff
@@ -232,6 +232,30 @@ public class Peripheral: NSObject, CBPeripheralDelegate {
         self.cbPeripheral.delegate = nil
     }
 
+    func updateDiscovery(_ peripheral: CBPeripheralInjectable, advertisements: [String : Any], RSSI: Int) throws {
+        guard self.identifier == peripheral.identifier else {
+            throw PeripheralError.invalidCbPeripheral
```

Comment:
If this happens something is really screwed up. Need to think if error should be exposed. Probably would want to recreate CBCentralManager.  Should fail scan and stop.